app/src/main/java/com/vultisig/wallet/ui/models/keysign/BuildKeysignTransactionUiModelUseCase.kt

```
@@ -0,0 +1,114 @@
+
+        val isDeposit =
+            when (val specific = keysignPayload.blockChainSpecific) {
+                is BlockChainSpecific.MayaChain -> specific.isDeposit
+                is BlockChainSpecific.THORChain -> specific.isDeposit
+                is BlockChainSpecific.Ton -> specific.isDeposit
+                is BlockChainSpecific.Cosmos ->
+                    specific.transactionType == TransactionType.TRANSACTION_TYPE_IBC_TRANSFER ||
+                        try {
+                            depositTransactionRepository.getTransaction(transactionId)
+                            true
+                        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)